dbt/include/athena/macros/materializations/models/table/table_hive_ha.sql

```diff
@@ -53,10 +51,8 @@
     -- swap table
     {% set swap_table = adapter.swap_table(tmp_relation.schema, tmp_relation.name, target_relation.schema, target_relation.table) %}
 
-    -- delete glue tmp table, do not use drop_relation, as it will remove data of the target table
-    {% call statement('drop_tmp_relation', auto_begin=False) -%}
-      drop table if exists {{ tmp_relation.render_hive() }}
-    {%- endcall %}
+    -- delete glue tmp table. drop_relation has logic such that table data will NOT be removed for a hight availability table
+    {{ drop_relation(tmp_relation) }}
```

Comment:
To avoid the `drop table` statement I've added logic in `drop_relation` to not delete table data if it's a high availability table. Let me know if this makes sense, otherwise I'll remove it.

Reply:
The idea here is that the future tmp relation is dropped before proceeding with table ha materialization.
Let's keep table ha logic in table materialization hive_table_ha and keep the drop_relation cleaner.
Therefore I suggest to use just a glue api call to delete the table.

Reply:
> plus https://github.com/dbt-athena/dbt-athena/pull/249/files#r1175697315 The proposed solution won't work
Ah, I see what you mean, there would also need to be a check on if it's a full refresh. I've reverted the additional hive_table_ha changes.